tests/external_botocore/test_bedrock_chat_completion.py

```diff
@@ -416,5 +440,366 @@ def _test():
+            with pytest.raises(json.decoder.JSONDecodeError):
+                json.loads(chunk["chunk"]["bytes"])
+
+    _test()
+
+
+_headers_error = botocore.eventstream.InvalidHeadersLength
+_headers_error_name = "botocore.eventstream:InvalidHeadersLength"
+
+
+@reset_core_stats_engine()
+def test_bedrock_chat_completion_error_malformed_response_streaming_chunk(
+    bedrock_server,
+    set_trace_info,
+):
+    """
+    After a streaming call, the chunk returned by the server is invalid and cannot be parsed.
+
+    * The library should throw an appropriate error, in this case InvalidHeadersLength.
+    * The thrown error will not include a response attribute as the response was invalid.
+    * No response can be parsed, meaning all response data should be missing.
+    """
+
+    @validate_custom_events(chat_completion_expected_malformed_response_streaming_chunk_events)
+    @validate_custom_event_count(count=2)
+    @validate_error_trace_attributes(
+        _headers_error_name,
+        exact_attrs={
+            "agent": {},
+            "intrinsic": {},
+            "user": {
+                "llm.conversation_id": "my-awesome-id",
+            },
+        },
+        forgone_params={
+            "agent": (),
+            "intrinsic": (),
+            "user": ("http.statusCode", "error.message", "error.code"),
+        },
+    )
+    @validate_transaction_metrics(
+        name="test_bedrock_chat_completion",
+        scoped_metrics=[("Llm/completion/Bedrock/invoke_model_with_response_stream", 1)],
+        rollup_metrics=[("Llm/completion/Bedrock/invoke_model_with_response_stream", 1)],
+        custom_metrics=[
+            ("Supportability/Python/ML/Bedrock/%s" % BOTOCORE_VERSION, 1),
+        ],
+        background_task=True,
+    )
+    @background_task(name="test_bedrock_chat_completion")
+    def _test():
+        model = "amazon.titan-text-express-v1"
+        body = (chat_completion_payload_templates[model] % ("Malformed Streaming Chunk", 0.7, 100)).encode("utf-8")
+        with pytest.raises(_headers_error):
+            set_trace_info()
+            add_custom_attribute("llm.conversation_id", "my-awesome-id")
+            add_custom_attribute("llm.foo", "bar")
+            add_custom_attribute("non_llm_attr", "python-agent")
+
+            response = bedrock_server.invoke_model_with_response_stream(
+                body=body,
+                modelId=model,
+                accept="application/json",
+                contentType="application/json",
+            )
+            response = "".join(chunk for chunk in response["body"])
+            assert response
+
+    _test()
+
+
+_event_stream_error = botocore.exceptions.EventStreamError
+_event_stream_error_name = "botocore.exceptions:EventStreamError"
+
+
+@reset_core_stats_engine()
+def test_bedrock_chat_completion_error_streaming_exception(
+    bedrock_server,
+    set_trace_info,
+):
+    """
+    During a streaming call, the chunk's headers indicate an exception.
```

Comment:
Re-writing another one as a second example:
```
During a streaming call, the streamed chunk's headers indicate an error. The streamed chunk's response body contains the error information from the server. The library uses the streamed chunk's error information to raise an actual exception. The response attribute on the raised exception will contain the error information. This means all response message data will be missing from the recorded events since the server returned an error instead of message data inside the streamed response.
```